packages/node/src/plugins/segmentio/publisher.ts

```diff
@@ -96,15 +117,19 @@ export class Publisher {
       and is always sent before a new batch is created.
 
       Add an event to the existing batch.
-        Success: Check if batch is full and send if it is.
+        Success: Check if batch is full or no more items are expected to come in (i.e. closing). If so, send batch.
         Failure: Assume event is too big to fit in current batch - send existing batch.
           Add an event to the new batch.
             Success: Check if batch is full and send if it is.
             Failure: Event exceeds maximum size (it will never fit), fail the event.
     */
 
     if (batch.tryAdd(pendingItem)) {
-      if (batch.length === this._maxEventsInBatch) {
+      const isFull = batch.length === this._maxEventsInBatch
+      const isExpectingNoMoreItems =
```

Comment:
I think you could do the `isExpectingNoMoreItems` check outside of this if statement, then re-use it here:
> if (fallbackBatch.length === this._maxEventsInBatch) {

That way it adding the last event to the batch forces a new batch (because the size in bytes is too large to fit in the existing one), we can still send it quickly.